src/Compilers/CSharp/Portable/Symbols/DelegateCaches/TypeParameterUsageChecker.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    using TypeParams = ImmutableArray<TypeParameterSymbol>;
+
+    /// <summary>
+    /// Checks whether a type or a method is constructed with some type that involves the specified generic type parameters.
+    /// </summary>
+    internal class TypeParameterUsageChecker : CSharpSymbolVisitor<TypeParams, bool>
```

Comment:
>TypeParameterUsageChecker [](start = 19, length = 25)

See `TypeSymbolExtensions.ContainsTypeParameter` to see how this could be done in one line.